src/foam/mlang/sink/TopNGroupBy.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * @license
+ * Copyright 2025 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.mlang.sink',
+  name: 'TopNGroupBy',
+  extends: 'foam.mlang.sink.GroupBy',
+
+  documentation: 'GroupBy sink that keeps only top N groups, aggregating others. works with only value-based grouping. (min, max, sum, avg, count)',
+
+  requires: [
+    'foam.mlang.sink.GroupBySortOrder'
+  ],
+
+  properties: [
+    {
+      class: 'Int',
+      name: 'topN',
+      value: 0,
+      documentation: 'Number of top groups to keep'
+    },
+    {
+      class: 'Enum',
+      of: 'foam.mlang.sink.GroupBySortOrder',
+      name: 'sortOrder',
+      value: 'DESC',
+      documentation: 'Sort order for value-based limiting'
+    },
+    {
+      class: 'String',
+      name: 'othersLabel',
+      value: 'Others',
+      documentation: 'Label for the aggregated "Others" category'
+    },
+    {
+      class: 'Boolean',
+      name: 'includeOthers',
+      value: true,
+      documentation: 'Whether to include "Others" group for remaining items'
+    }
+  ],
+
+  static: [
+    {
+      name: 'isSupported',
+      args: 'foam.dao.Sink sink',
+      type: 'boolean',
+      code: function isSupported(sink) {
+        return foam.mlang.sink.Sum.isInstance(sink) ||
+                foam.mlang.sink.Average.isInstance(sink) ||
+               foam.mlang.sink.Count.isInstance(sink) ||
+               foam.mlang.sink.Min.isInstance(sink) ||
+               foam.mlang.sink.Max.isInstance(sink);
+      },
+      javaCode: `
```

Comment:
See Ruby's changes in her Pivot PR. But you'll need to wait until we merge that.

Reply:
integrated in https://github.com/kgrgreer/foam3/pull/4261